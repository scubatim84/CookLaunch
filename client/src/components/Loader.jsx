import React from 'react';
import {useStylesMain} from '../Styles';
import {themeMain} from '../Theme';
import {Backdrop, CircularProgress} from '@material-ui/core';

function Loader() {
  const classes = useStylesMain(themeMain);

  return (
    <div className={classes.minHeight}>
      <Backdrop className={classes.backdrop} open={true}>
        <CircularProgress color='inherit' />
      </Backdrop>
    </div>
  );
}

export default Loader;
