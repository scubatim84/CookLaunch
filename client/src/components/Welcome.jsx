import React from 'react';
import { Redirect } from 'react-router-dom';
import {
  Backdrop,
  CircularProgress,
  Grid,
  Typography,
} from '@material-ui/core';

import { useStylesMain } from '../Styles';
import { themeMain } from '../Theme';
import Loader from './Loader';

const Welcome = (props) => {
  const classes = useStylesMain(themeMain);

  if (!props.isLoggedIn) {
    return <Redirect to='/login' />;
  }
  if (!props.firstName) {
    return <Loader />;
  }

  return (
    <div className={classes.minHeight}>
      <Backdrop className={classes.backdrop} open={!props.firstName}>
        <CircularProgress color='inherit' />
      </Backdrop>
      <Grid container spacing={2} className={classes.pagePadding}>
        <Grid item xs={12}>
          <Typography id='FirstName' variant='h4' color='textPrimary'>
            Welcome {props.firstName}!
          </Typography>
        </Grid>
        <Grid item xs={12}>
          <Typography variant='h6' color='textPrimary'>
            You currently have no recipes. To get started, please follow these
            three steps using the profile menu in the top right part of the
            screen:
          </Typography>
        </Grid>
        <Grid item xs={12}>
          <Typography variant='h6' color='textPrimary'>
            1) Add ingredients desired for your first recipe
          </Typography>
        </Grid>
        <Grid item xs={12}>
          <Typography variant='h6' color='textPrimary'>
            2) Add your first recipe
          </Typography>
        </Grid>
        <Grid item xs={12}>
          <Typography variant='h6' color='textPrimary'>
            3) Once you have added a recipe, view it here on your dashboard
          </Typography>
        </Grid>
      </Grid>
    </div>
  );
};

export default Welcome;
