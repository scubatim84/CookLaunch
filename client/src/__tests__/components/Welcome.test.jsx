import React from 'react';
import { shallow } from 'enzyme';
import Welcome from '../../components/Welcome';
import { Redirect } from 'react-router-dom';
import Loader from '../../components/Loader';
import {
  Backdrop,
  CircularProgress,
  Grid,
  Typography,
} from '@material-ui/core';

describe('Welcome', () => {
  it('Renders component without crashing', () => {
    shallow(<Welcome />);
  });

  it('Renders redirect component to login route if not logged in', () => {
    const wrapper = shallow(<Welcome />);

    expect(wrapper.find(Redirect)).toHaveLength(1);
  });

  it('Renders loader component if there is no first name prop', () => {
    const wrapper = shallow(<Welcome isLoggedIn={true} />);

    expect(wrapper.find(Loader)).toHaveLength(1);
  });

  it('Renders welcome component if logged in and first name provided', () => {
    const wrapper = shallow(<Welcome isLoggedIn={true} firstName='Tim' />);

    expect(wrapper.find(Backdrop)).toHaveLength(1);
    expect(wrapper.find(CircularProgress)).toHaveLength(1);
    expect(wrapper.find(Grid)).toHaveLength(6);
    expect(wrapper.find(Typography)).toHaveLength(5);
    expect(wrapper.find('#FirstName').equals('Welcome Tim'));
  });
});
