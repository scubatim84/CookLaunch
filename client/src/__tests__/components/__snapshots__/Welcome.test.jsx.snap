import React from 'react';
import {shallow} from 'enzyme';
import Welcome from '../../components/Welcome';

describe('Welcome', () => {
  it('Renders correctly', () => {
    const wrapper = shallow(<Welcome />);

    expect(wrapper).toMatchSnapshot();
  });
});
